### PR TITLE
Next fixes

### PR DESCRIPTION
**Pull request by martijn00**
@ebozduman

### CODE REVIEW

File: Minio/Credentials/MinioEnvironmentProvider.cs [hunk 1]
```diff
@@ -19,19 +19,17 @@
...
 
-public class MinioEnvironmentProvider : EnvironmentProvider
+public class MinioEnvironmentProvider : IClientProvider
 {
-    public override AccessCredentials GetCredentials()
+    public AccessCredentials GetCredentials()
     {
-        var credentials = new AccessCredentials(GetEnvironmentVariable("MINIO_ACCESS_KEY"),
-            GetEnvironmentVariable("MINIO_SECRET_KEY"), null, default);
+        var credentials = new AccessCredentials(Environment.GetEnvironmentVariable("MINIO_ACCESS_KEY"),
```
Comment (ebozduman) [inline Minio/Credentials/MinioEnvironmentProvider.cs:26]: `MINIO_ACCESS_KEY` and `MINIO_SECRET_KEY` are deprecated.
The newly coming env vars are `MINIO_ROOT_USER` and `MINIO_ROOT_PASSWORD`.
However, we still honor the deprecated ones if the new env vars are not defined.

**Review by ebozduman — CHANGES_REQUESTED** [review]

**Review by ebozduman — APPROVED** [review]
LGTM

**Comment by martijn00** [comment]
@ebozduman I don't know why the test fails. Locally this test passes. Can you try run it again?

**Comment by martijn00** [comment]
@ebozduman I've done a rebase to include the parallel tests. Can you run it again?

**Comment by ebozduman** [comment]
I get the same result; just cannot reproduce the failure reported during build tests on the windows side.
I'll dig in more.

**Comment by martijn00** [comment]
@ebozduman any news?

**Pull request merged**